.travis.yml

```diff
@@ -79,14 +104,21 @@ script:
   # to check that this works fine.
   - cargo check --verbose --all-targets
 
-  - cargo build --verbose --all-targets --all-features
-  - cargo test --verbose --all-features
-  - cargo doc --verbose --all-features
+  # Skip XCBConnection on cross (libxcb not available)
+  - |
+    if [ -z "$CROSS_TARGET" ]; then
+        export FEATURES="--all-features"
+      else
+        export FEATURES="--features all-extensions"
+      fi
+  - cargo build --verbose --all-targets $FEATURES
+  - cargo test --verbose $FEATURES
+  - cargo doc --verbose $FEATURES
 
   # Run the examples as 'integration tests'.
   # Enable the 'all-extensions' feature or Cargo will complain.
   # Enable the 'allow-unsafe-code' feature so XCBConnection is used.
-  - run_examples --all-features
+  - skip_on_cross run_examples --all-features
 
   - cargo build --verbose --all-targets --features all-extensions
```

Comment:
This looks like it does `cargo build --verbose --all-targets --features all-extensions` twice.
Maybe MIPS CI could be reduced to just run parsing tests?

Reply:
> This looks like it does `cargo build --verbose --all-targets --features all-extensions` twice.

Well, that should not cause slowdown, the second time the command would just finish immediately.

Reply:
> Maybe MIPS CI could be reduced to just run parsing tests?

As in something like this? This now has `skip_on_cross` on everything but the call to `cargo test`.
```diff
diff --git a/.travis.yml b/.travis.yml
index d332c4c..b2ff68f 100644
--- a/.travis.yml
+++ b/.travis.yml
@@ -94,7 +94,7 @@ before_script:
 script:
   # Check the code generator. Output should be identical to the
   # committed version.
-  - make
+  - skip_on_cross make
   - if ! git diff --exit-code; then exit 1; fi
 
   - run_clippy
@@ -102,7 +102,7 @@ script:
 
   # Build once with just the default features, i.e. without all the extensions
   # to check that this works fine.
-  - cargo check --verbose --all-targets
+  - skip_on_cross cargo check --verbose --all-targets
 
   # Skip XCBConnection on cross (libxcb not available)
   - |
@@ -111,16 +111,16 @@ script:
       else
         export FEATURES="--features all-extensions"
       fi
-  - cargo build --verbose --all-targets $FEATURES
+  - skip_on_cross cargo build --verbose --all-targets $FEATURES
   - cargo test --verbose $FEATURES
-  - cargo doc --verbose $FEATURES
+  - skip_on_cross cargo doc --verbose $FEATURES
 
   # Run the examples as 'integration tests'.
   # Enable the 'all-extensions' feature or Cargo will complain.
   # Enable the 'allow-unsafe-code' feature so XCBConnection is used.
   - skip_on_cross run_examples --all-features
 
-  - cargo build --verbose --all-targets --features all-extensions
+  - skip_on_cross cargo build --verbose --all-targets --features all-extensions
 
   # Run the examples as 'integration tests'. This time using RustConnection.
-  - run_examples --features "all-extensions libc"
+  - skip_on_cross run_examples --features "all-extensions libc"
```

Reply:
Note that in Travis you can have an `include` entry use a specific `script`:

```yaml
    - env: CROSS_TARGET=mips64-unknown-linux-gnuabi64 CLIPPY_RUSTFMT=no
      rust: stable
      services: docker
      sudo: required 
      script:
        - cross test --target=...
```

Reply:
Nice, I didn't know that.